pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -60,6 +60,12 @@ func (r *pullthroughBlobStore) remoteStat(ctx context.Context, dgst digest.Diges
 		return distribution.Descriptor{}, err
 	}
 
+	r.pullFromInsecureRegistries = false
```

Comment:
I also wanted to protect against possible changes that I made in a previous version of this PR.